### PR TITLE
cargo fmt rust code

### PR DESCRIPTION
**Pull request by rongyi**
using cargo tool to format rust code

### CODE REVIEW

File: codes/rust/chapter_computational_complexity/time_complexity.rs [hunk 1]
```diff
@@ -49,9 +49,9 @@ fn quadratic(n: i32) -> i32 {
 /* 平方阶（冒泡排序） */
 fn bubble_sort(nums: &mut [i32]) -> i32 {
     let mut count = 0; // 计数器
-    // 外循环：未排序区间为 [0, i]
+                       // 外循环：未排序区间为 [0, i]
```
Comment (krahets) [inline codes/rust/chapter_computational_complexity/time_complexity.rs:52]: Incorrect indentation. (Many others too)

Comment (diucicd) [reply codes/rust/chapter_computational_complexity/time_complexity.rs]: oh, need to add an empty line to two unrelated comment lines. code updated! please review

Comment (rongyi) [reply codes/rust/chapter_computational_complexity/time_complexity.rs]: oh, need to add an empty line to separate unrelated comment line, code updated

Comment (rongyi) [reply codes/rust/chapter_computational_complexity/time_complexity.rs]: also find some other warnings, maybe better in another PR

File: codes/rust/chapter_sorting/merge_sort.rs [hunk 2]
```diff
@@ -43,12 +43,14 @@ fn merge(nums: &mut [i32], left: usize, mid: usize, right: usize) {
...
-    merge_sort(nums, mid + 1, right);  // 递归右子数组
-    // 合并阶段
+    if left >= right {
+        return;
+    } // 当子数组长度为 1 时终止递归
+      // 划分阶段
+    let mid = (left + right) / 2; // 计算中点
+    merge_sort(nums, left, mid); // 递归左子数组
+    merge_sort(nums, mid + 1, right); // 递归右子数组
+                                      // 合并阶段
```
Comment (krahets) [inline codes/rust/chapter_sorting/merge_sort.rs:53]: indent

File: codes/rust/chapter_sorting/bubble_sort.rs [hunk 3]
```diff
@@ -27,29 +27,32 @@ fn bubble_sort_with_flag(nums: &mut [i32]) {
...
                 nums[j] = nums[j + 1];
                 nums[j + 1] = tmp;
-                flag = true;  // 记录交换元素
+                flag = true; // 记录交换元素
             }
         }
-        if !flag {break};  // 此轮“冒泡”未交换任何元素，直接跳出
+        if !flag {
+            break;
+        }; // 此轮“冒泡”未交换任何元素，直接跳出
```
Comment (krahets) [inline codes/rust/chapter_sorting/bubble_sort.rs:43]: move this comment to line 42

File: codes/rust/chapter_computational_complexity/time_complexity.rs (hunk 1 above)
Comment (krahets) [reply codes/rust/chapter_computational_complexity/time_complexity.rs]: Sounds great!

File: codes/rust/chapter_sorting/merge_sort.rs [hunk 4]
```diff
@@ -43,12 +43,14 @@ fn merge(nums: &mut [i32], left: usize, mid: usize, right: usize) {
...
-    if left >= right { return; }             // 当子数组长度为 1 时终止递归
-    // 划分阶段
-    let mid = (left + right) / 2;     // 计算中点
-    merge_sort(nums, left, mid);      // 递归左子数组
-    merge_sort(nums, mid + 1, right);  // 递归右子数组
-    // 合并阶段
+    if left >= right {
+        return;
+    } // 当子数组长度为 1 时终止递归
+      // 划分阶段
```
Comment (krahets) [inline codes/rust/chapter_sorting/merge_sort.rs:49]: indent

File: codes/rust/chapter_sorting/merge_sort.rs [hunk 5]
```diff
@@ -43,12 +43,14 @@ fn merge(nums: &mut [i32], left: usize, mid: usize, right: usize) {
...
     // 终止条件
-    if left >= right { return; }             // 当子数组长度为 1 时终止递归
-    // 划分阶段
-    let mid = (left + right) / 2;     // 计算中点
-    merge_sort(nums, left, mid);      // 递归左子数组
-    merge_sort(nums, mid + 1, right);  // 递归右子数组
-    // 合并阶段
+    if left >= right {
+        return;
+    } // 当子数组长度为 1 时终止递归
```
Comment (krahets) [inline codes/rust/chapter_sorting/merge_sort.rs:48]: move the comment to line 47

Please check others as well

File: codes/rust/chapter_sorting/merge_sort.rs (hunk 2 above)
Comment (rongyi) [reply codes/rust/chapter_sorting/merge_sort.rs]: updated

File: codes/rust/chapter_sorting/merge_sort.rs (hunk 4 above)
Comment (rongyi) [reply codes/rust/chapter_sorting/merge_sort.rs]: must say, such a great owner!

File: codes/rust/chapter_sorting/merge_sort.rs (hunk 5 above)
Comment (rongyi) [reply codes/rust/chapter_sorting/merge_sort.rs]: updated

File: codes/rust/chapter_sorting/bubble_sort.rs [hunk 6]
```diff
@@ -27,29 +27,33 @@ fn bubble_sort_with_flag(nums: &mut [i32]) {
...
                 // 交换 nums[j] 与 nums[j + 1]
                 let tmp = nums[j];
                 nums[j] = nums[j + 1];
                 nums[j + 1] = tmp;
-                flag = true;  // 记录交换元素
+                flag = true; // 记录交换元素
             }
         }
-        if !flag {break};  // 此轮“冒泡”未交换任何元素，直接跳出
+        // 此轮“冒泡”未交换任何元素，直接跳出
```
Comment (krahets) [inline codes/rust/chapter_sorting/bubble_sort.rs:41]: We should place this comment to line 43 (after break;)/

If placed here, it should be "如果此轮“冒泡”未交换任何元素，则直接跳出"

Same as others

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**